guardian/shortcuts.py

```diff
@@ -159,7 +170,7 @@ def get_perms_for_model(cls):
     return Permission.objects.filter(content_type=ctype)
 
 def get_users_with_perms(obj, attach_perms=False, with_superusers=False,
-        with_group_users=True):
+        with_group_users=True, direct_perms_only=False):
```

Comment:
#403, #407 do solve my issue. I think the new get_users_with_perms() behavior is great and what I originally expected (see #327). However, it is a backwards-incompatible change and so may break existing apps that depend on the previous behavior. Probably worth calling out in the release notes.

 Arguably, the new behavior could be considered a bug fix, but there may be applications depending on the new behavior that will break when upgraded.